addons/hr_timesheet/static/src/components/time_hour_field/time_hour_field.js

```diff
@@ -0,0 +1,19 @@
+import { registry } from "@web/core/registry";
+import { FloatTimeField } from "@web/views/fields/float_time/float_time_field";
+
+export class TimeHourField extends FloatTimeField {
+    get formattedValue() {
+        const unitAmount = super.formattedValue;
+        const [hours, minutes] = unitAmount.split(":").map(Number);
+        let durationDisplay = `${hours}h`;
+        if (minutes > 0) {
+            durationDisplay += ` ${minutes}m`;
+        }
```

Comment:
here we are casting things from strings to numbers and then back to strings - that shouldn't be necessary 
```suggestion
        const [hours, minutes] = unitAmount.split(":");
        let durationDisplay = hours + 'h';
        if (minutes !== '00') {
            durationDisplay += minutes + 'm';
        }
```